fastlane/fastlane
# [action][swiftlint] Adds '--progress' flag

### Checklist


### Motivation and Context

I think it is a good idea to support the full feature set of a used tool, in this case SwiftLint, inside the fastlane actions. This pull request is a contribution to achieve that goal.
The newly introduced flag `--progress` of SwiftLint helps to reduce clutter from the console output. The output is reduced to SwiftLint violations only which makes it much easier to find those inside a console log be it on a local machine or on a CI.

### Description

I added a new accepted optional argument to the fastlane action 'swiftlint' that will add the '--progress' flag to the SwiftLint command if passed by the action with `true`. If the `--progress` flag is not supported by the current SwiftLint version (first introduced with 0.49.1) the action argument will be ignored. The default value for that argument is `false` to not introduce a different behaviour to previous fastlane versions.

### Testing Steps

1. Clone the repo, checkout the PR branch
2. CD into the directory and execute `bundle install` and/or `bundle update` (You should have installed `bundler` already)
3. CD into you project and edit your projects `Gemfile` to use `gemspec path: File.expand_path("<PATH_TO_LOCAL_FASTLANE>") instead of `gem fastlane`; replacing <PATH_TO_LOCAL_FASTLANE> with the path where you have the local fastlane repo.
5. Execute `bundle exec fastlane run swiftlint progress:true` to see how it works.
